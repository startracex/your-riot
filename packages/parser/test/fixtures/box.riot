<content class="container">
  <div class="box" Align="center">
    <h1>{box.title}</h1>
    <img Src={box.image} width="480" />
    <div class="body">
      <h2>Escaped \{0}</h2>
      <p>Body: {box.body}</p>
      <pre>
        Do not trim <span>this</span>
      </pre>
    </div>
  </div>
  <p/>

  <style type="less">
    pre {
      span: { font-weight: bolder }
    }
  </style>

  <script>
  this.box = {
      title: "Good morning!",
      image: "http://trinixy.ru/pics5/20130614/podb_07.jpg",
      body: "It is when SO!"
  }
  </script>

</content>
