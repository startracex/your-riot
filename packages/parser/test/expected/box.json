{
  "template": {
    "type": 1,
    "name": "content",
    "start": 0,
    "end": 559,
    "isCustom": true,
    "attributes": [
      {
        "name": "class",
        "value": "container",
        "start": 9,
        "end": 26,
        "valueStart": 16,
        "parts": [
          "container"
        ]
      }
    ],
    "nodes": [
      {
        "type": 1,
        "name": "div",
        "start": 30,
        "end": 291,
        "attributes": [
          {
            "name": "class",
            "value": "box",
            "start": 35,
            "end": 46,
            "valueStart": 42,
            "parts": [
              "box"
            ]
          },
          {
            "name": "Align",
            "value": "center",
            "start": 47,
            "end": 61,
            "valueStart": 54,
            "parts": [
              "center"
            ]
          }
        ],
        "nodes": [
          {
            "type": 1,
            "name": "h1",
            "start": 67,
            "end": 87,
            "nodes": [
              {
                "type": 3,
                "text": "{box.title}",
                "start": 71,
                "end": 82,
                "expressions": [
                  {
                    "text": "box.title",
                    "start": 71,
                    "end": 82
                  }
                ],
                "parts": [
                  "box.title"
                ]
              }
            ]
          },
          {
            "type": 1,
            "name": "img",
            "start": 92,
            "end": 127,
            "isVoid": true,
            "attributes": [
              {
                "name": "Src",
                "value": "{box.image}",
                "start": 97,
                "end": 112,
                "expressions": [
                  {
                    "text": "box.image",
                    "start": 101,
                    "end": 112
                  }
                ],
                "valueStart": 101,
                "parts": [
                  "box.image"
                ]
              },
              {
                "name": "width",
                "value": "480",
                "start": 113,
                "end": 124,
                "valueStart": 120,
                "parts": [
                  "480"
                ]
              }
            ],
            "isSelfClosing": true
          },
          {
            "type": 1,
            "name": "div",
            "start": 132,
            "end": 282,
            "attributes": [
              {
                "name": "class",
                "value": "body",
                "start": 137,
                "end": 149,
                "valueStart": 144,
                "parts": [
                  "body"
                ]
              }
            ],
            "nodes": [
              {
                "type": 1,
                "name": "h2",
                "start": 157,
                "end": 178,
                "nodes": [
                  {
                    "type": 3,
                    "text": "Escaped \\{0}",
                    "start": 161,
                    "end": 173,
                    "unescape": "{",
                    "parts": [
                      "Escaped {0}"
                    ]
                  }
                ]
              },
              {
                "type": 1,
                "name": "p",
                "start": 185,
                "end": 208,
                "nodes": [
                  {
                    "type": 3,
                    "text": "Body: {box.body}",
                    "start": 188,
                    "end": 204,
                    "expressions": [
                      {
                        "text": "box.body",
                        "start": 194,
                        "end": 204
                      }
                    ],
                    "parts": [
                      "Body: ",
                      "box.body"
                    ]
                  }
                ]
              },
              {
                "type": 1,
                "name": "pre",
                "start": 215,
                "end": 271,
                "isRaw": true,
                "nodes": [
                  {
                    "type": 3,
                    "text": "\n        Do not trim ",
                    "start": 220,
                    "end": 241,
                    "parts": [
                      "\\n        Do not trim "
                    ]
                  },
                  {
                    "type": 1,
                    "name": "span",
                    "start": 241,
                    "end": 258,
                    "isRaw": true,
                    "nodes": [
                      {
                        "type": 3,
                        "text": "this",
                        "start": 247,
                        "end": 251,
                        "parts": [
                          "this"
                        ]
                      }
                    ]
                  },
                  {
                    "type": 3,
                    "text": "\n      ",
                    "start": 258,
                    "end": 265,
                    "parts": [
                      "\\n      "
                    ]
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        "type": 1,
        "name": "p",
        "start": 294,
        "end": 298,
        "isSelfClosing": true
      }
    ]
  },
  "css": {
    "type": 1,
    "name": "style",
    "start": 302,
    "end": 384,
    "attributes": [
      {
        "name": "type",
        "value": "less",
        "start": 309,
        "end": 320,
        "valueStart": 315,
        "parts": [
          "less"
        ]
      }
    ],
    "text": {
      "type": 3,
      "text": "\n    pre {\n      span: { font-weight: bolder }\n    }\n  ",
      "start": 321,
      "end": 376
    }
  },
  "javascript": {
    "type": 1,
    "name": "script",
    "start": 388,
    "end": 547,
    "text": {
      "type": 3,
      "text": "\n  this.box = {\n      title: \"Good morning!\",\n      image: \"http://trinixy.ru/pics5/20130614/podb_07.jpg\",\n      body: \"It is when SO!\"\n  }\n  ",
      "start": 396,
      "end": 538
    }
  }
}